csrc/fusion.cpp

```diff
@@ -313,7 +313,7 @@ bool Fusion::isNoOp() {
     auto root_dom = TensorDomain::noReductions(out_tv->getMaybeRFactorDomain());
     bool size_zero = false;
     for (auto id : root_dom) {
-      if (id->extent()->isZeroInt()) {
+      if (id->extent()->isConstScalar() && id->extent()->evaluateInt() == 0) {
```

Comment:
Can `Val::isZero()` be extended to support the index type as well?

Reply:
`isZero` already supports index type. The problem here is, not all zero extents are constant zero. It can be, for example `castOp(DataType::Index, 0)` or `5 - 5`, etc. And we should evaluate them to know if it is actually zero.